scraper.rb

```diff
@@ -5,66 +5,152 @@
 require 'nokogiri'
 require 'colorize'
 require 'pry'
-require 'open-uri/cached'
-OpenURI::Cache.cache_path = '.cache'
+# require 'open-uri/cached'
+# OpenURI::Cache.cache_path = '.cache'
+require 'scraped_page_archive/open-uri'
+
+module NokoHelper
+  def noko_for(url)
+    @noko ||= Nokogiri::HTML(open(url).read)
+  end
+
+  def noko
+    noko_for(url)
+  end
+end
 
 class String
   def tidy
-    self.gsub(/[[:space:]]+/, ' ').strip
+    gsub(/[[:space:]]+/, ' ').strip
   end
 end
 
-def noko_for(url)
-  Nokogiri::HTML(open(url).read)
-end
+class List
+  include NokoHelper
 
-def dob_from(node)
-  Date.parse(node.text.tidy[/Born\s+(?:on)\s+(\d+\s+\w+\s+\d+)/, 1]).to_s rescue ''
-end
+  def initialize(term, url)
+    @term = term
+    @url = url
+  end
 
-def scrape_list(term, url)
-  noko = noko_for(url)
-  noko.css('.liste2 .liste a').each do |a|
-    link = URI.join url, a.attr('href')
-    scrape_mp(term, a.text, link)
+  def members
```

Comment:
This method seems to be doing quite a lot. I'd suggest splitting the various parts of this up a bit more.